ext/mbstring/mbstring.c

```diff
@@ -1763,24 +1763,20 @@ PHP_FUNCTION(mb_http_input)
  	if (typ == NULL) {
  		result = MBSTRG(http_input_identify);
  	} else {
+		*typ |= 0x20;
```

Comment:
A comment explaining this bit field would be helpful for `mbstring` maintainers

Reply:
You can't modify `*typ` here, because you don't own the string. This is going to cause memory corruption for interned strings.

Generally I don't think we should make this change. This code is not performance-critical, and the existing code is quite explicit.

Reply:
how about this version?